plugins/search/src/components/LegacySearchPage/LegacySearchResult.tsx

```diff
@@ -0,0 +1,291 @@
+  searchQuery: {
+    color: theme.palette.text.primary,
+    background: theme.palette.background.default,
+    borderRadius: '10%',
+  },
+  tableHeader: {
+    margin: theme.spacing(1, 0, 0, 0),
+    display: 'flex',
+  },
+  divider: {
+    width: '1px',
+    margin: theme.spacing(0, 2),
+    padding: theme.spacing(2, 0),
+  },
+}));
+
+type SearchResultProps = {
+  searchQuery?: string;
+};
+
+type TableHeaderProps = {
+  searchQuery?: string;
+  numberOfSelectedFilters: number;
+  numberOfResults: number;
+  handleToggleFilters: () => void;
+};
+
+// TODO: move out column to make the search result component more generic
+const columns: TableColumn[] = [
+  {
+    title: 'Name',
+    field: 'name',
+    highlight: true,
+    render: (result: Partial<Result>) => (
+      <Link to={result.url || ''}>{result.name}</Link>
+    ),
+  },
+  {
+    title: 'Description',
+    field: 'description',
+  },
+  {
+    title: 'Owner',
+    field: 'owner',
+  },
+  {
+    title: 'Kind',
+    field: 'kind',
+  },
+  {
+    title: 'LifeCycle',
+    field: 'lifecycle',
+  },
+];
+
+const TableHeader = ({
+  searchQuery,
+  numberOfSelectedFilters,
+  numberOfResults,
+  handleToggleFilters,
+}: TableHeaderProps) => {
+  const classes = useStyles();
+
+  return (
+    <div className={classes.tableHeader}>
+      <FiltersButton
+        numberOfSelectedFilters={numberOfSelectedFilters}
+        handleToggleFilters={handleToggleFilters}
+      />
+      <Divider className={classes.divider} orientation="vertical" />
+      <Grid item xs={12}>
+        {searchQuery ? (
+          <Typography variant="h6">
+            {`${numberOfResults} `}
+            {numberOfResults > 1 ? `results for ` : `result for `}
+            <span className={classes.searchQuery}>"{searchQuery}"</span>{' '}
+          </Typography>
+        ) : (
+          <Typography variant="h6">{`${numberOfResults} results`}</Typography>
+        )}
+      </Grid>
+    </div>
+  );
+};
+
+export const SearchResult = ({ searchQuery }: SearchResultProps) => {
+  const catalogApi = useApi(catalogApiRef);
+
+  const [showFilters, toggleFilters] = useState(false);
+  const [selectedFilters, setSelectedFilters] = useState<FiltersState>({
+    selected: '',
+    checked: [],
+  });
+
+  const [filteredResults, setFilteredResults] = useState<SearchResults>([]);
+
+  const { loading, error, value: results } = useAsync(async () => {
+    const entities = await catalogApi.getEntities();
+    return entities.items.map((entity: Entity) => ({
+      name: entity.metadata.name,
+      description: entity.metadata.description,
+      owner:
+        typeof entity.spec?.owner === 'string' ? entity.spec?.owner : undefined,
+      kind: entity.kind,
+      lifecycle:
+        typeof entity.spec?.lifecycle === 'string'
+          ? entity.spec?.lifecycle
+          : undefined,
+      url: `/catalog/${
+        entity.metadata.namespace?.toLowerCase() || ENTITY_DEFAULT_NAMESPACE
```

Comment:
May be postponed to after merge but still important: avoid all use of `toLowerCase`, replace with `toLocaleLowerCase('en-US')`. I see you're using it in a lot of other places, so maybe just some stragglers that remain. There are some nasty corner cases with for example Turkish ı, which is a dotless i